### PR TITLE
Fix CAIP-122 verification

### PR DESCRIPTION
**Pull request by wilwade**
# Problem

Various issues found when testing the CAIP-122 signature verification

# Solution

1. Fixed a few SIWA references
2. Fixed the signature verification to allow for the hashed version in addition to the pure string version.
3. `publicKey` should be `Sr` not `SR` (verification doesn't care however)
4. Added additional test generated from FA code
5. Match address encoding before comparing for the CAIP-122 payload
6. Fix address extraction
7. Fix expired parsing

### CODE REVIEW

File: docs/src/DataStructures/Request.md
Comment (wilwade) [inline docs/src/DataStructures/Request.md:1]: Generated

File: docs/src/DataStructures/RequestUrl.md
Comment (wilwade) [inline docs/src/DataStructures/RequestUrl.md:1]: Generated

File: docs/src/DataStructures/Response-LoginOnly.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
     "encodedValue": "f6akufkq9Lex6rT8RCEDRuoZQRgo5pWiRzeo81nmKNGWGNJdJ",
     "encoding": "base58",
     "format": "ss58",
-    "type": "SR25519"
+    "type": "Sr25519"
```
Comment (wilwade) [inline docs/src/DataStructures/Response-LoginOnly.md:7]: Switch is due to how FA is doing things. No really matter either way

File: libraries/js/src/payloads.ts [hunk 2]
```diff
@@ -20,14 +21,17 @@ interface SiwxMessage {
   nonce: string;
   expired: boolean;
   issuedAt: Date;
-  expirationTime: Date;
+  expirationTime?: Date;
```
Comment (wilwade) [inline libraries/js/src/payloads.ts:24]: Not required

File: libraries/js/src/payloads.ts [hunk 3]
```diff
@@ -54,6 +58,36 @@ function parseMessage(message: string): SiwxMessage {
   };
 }
 
+// SIWA is switching away from this, but we should still support it for now
+function verifySignatureHashMaybeWrapped(publicKey: string, signature: string, message: Uint8Array): boolean {
+  const unwrappedSigned = message.length > 256 ? blake2AsU8a(message) : message;
```
Comment (wilwade) [inline libraries/js/src/payloads.ts:63]: This is from how the Polkadot-SDK signs payloads. We might want to require it, but for now we support it either way.

File: libraries/js/src/response.test.ts [hunk 4]
```diff
@@ -84,4 +84,11 @@ describe('validateSiwfResponse', () => {
       'Response failed to correctly parse or invalid content: {"foo":"bad"}'
     );
   });
+
+  it('throws on a bad domain', async () => {
```
Comment (wilwade) [inline libraries/js/src/response.test.ts:88]: Extra test

File: libraries/js/src/payloads.ts [hunk 5]
```diff
@@ -54,6 +58,36 @@ function parseMessage(message: string): SiwxMessage {
...
+
+  const unwrappedVerifyResult = signatureVerify(unwrappedSigned, signature, publicKey);
+  if (unwrappedVerifyResult.isValid) {
+    return true;
+  }
+
+  // Support both wrapped and unwrapped signatures
+  const wrappedSignedBytes = u8aWrapBytes(message);
+  const wrappedSigned = wrappedSignedBytes.length > 256 ? blake2AsU8a(wrappedSignedBytes) : wrappedSignedBytes;
```
Comment (enddynayn) [inline libraries/js/src/payloads.ts:72]: 👍

**Review by enddynayn — APPROVED** [review]
👍 great!

**Pull request merged**